hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsRestOperation.java

```diff
@@ -326,4 +345,37 @@ private void incrementCounter(AbfsStatistic statistic, long value) {
       abfsCounters.incrementCounter(statistic, value);
     }
   }
+
+  private void updateRequestHeaders() {
```

Comment:
AbfsRestOperation is not supposed to have any API specific implementations or handling. Come up with a LeaseRetryPolicy that will be used when executing operations that pass in leaseId. That can control retry count to 2 or how many ever reduced number of retries needed. Post that, if the request still fails, retry policy should exit AbfsRestOperation and abfsClient should take care of retrying from the respective method post updating request headers.